earth_moving/clutter/engine2.py

```diff
@@ -0,0 +1,346 @@
+# this file contains the PyBulletEnvironment class
+# the general capabilities are opening the environment, closing the environment, and general control functions
+
+# general imports
+import pybullet as p
+import pybullet_data
+import time as t
+import numpy as np
+import math
+
+# class definition
+class PyBulletEnvironment:
+    
+    # constructor
+    def __init__(self, gui=True, gravity=(0, 0, -10)):
+        """
+        Initialize the PyBullet environment.
+
+        :param gui: Boolean, if True the environment will be opened with a GUI.
+        :param gravity: Tuple, the gravity vector.
+        """
+        
+        # gui
+        self.gui = gui
+        self.physicsClient = None 
+        
+        # control accuracy threshold        
+        self.control_threshold = 1e-2 
+        
+        # iter limit
+        self.iter_limit = 5e2
+        
+        # ID list 
+        self.ID = []
+        
+        # pybullet options
+        self.URDF_MERGE_FIXED_LINKS = True
```

Comment:
See my comment above on naming conventions. We sometimes use `UPPERCASE_WITH_UNDERSCORES`, but only for constants... Is this the case? Also, do we want to read those from a configuration file? Let's talk about that.

Reply:
actually these are all names from pybullet flags, so I just copied the name